pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -95,6 +96,7 @@ func NewDefaultProvider(ctx context.Context, cache *cache.Cache, ec2api ec2iface
 		cm:                    pretty.NewChangeMonitor(),
 		KubeDNSIP:             kubeDNSIP,
 		ClusterEndpoint:       clusterEndpoint,
+		ClusterIPFamily:       lo.Ternary(kubeDNSIP.To4() == nil, corev1.IPv6Protocol, corev1.IPv4Protocol),
```

Comment:
Will need to be nil-safe since `kubeDNSIP` can be `nil`?

Reply:
Is this for self-managed clusters?